### PR TITLE
Converted composition reaction materal model to modern interface.

### PR DESCRIPTION
**Pull request by jperryhouts**

### CODE REVIEW

**Comment by bangerth** [comment]
@tjhei -- why does the tester not run for this PR?

**Comment by tjhei** [comment]
I don't know why it is not reported, but it got tested: [1]. You can always check [2].

[1]: http://www.math.clemson.edu/~heister/aspect-logs/9b729353838acdb502400314b92b0d0414ff8b77/
[2]: http://www.math.clemson.edu/~heister/aspect-logs/results.html

File: source/material_model/composition_reaction.cc [hunk 1]
```diff
@@ -30,37 +30,78 @@ namespace aspect
...
-          case 1:
-            //geometric interpolation
-            return (pow(10, ((1-composition[0]) * log10(eta*temperature_dependence)
-                             + composition[0] * log10(eta*composition_viscosity_prefactor_1*temperature_dependence))));
-          default:
-            return (pow(10, ((1 - 0.5*composition[0] - 0.5*composition[1]) * log10(eta*temperature_dependence)
-                             + 0.5 * composition[0] * log10(eta*composition_viscosity_prefactor_1*temperature_dependence)
-                             + 0.5 * composition[1] * log10(eta*composition_viscosity_prefactor_2*temperature_dependence))));
+          const double temperature = in.temperature[i];
+          const std::vector<double> composition = in.composition[i];
```
Comment (tjhei) [inline source/material_model/composition_reaction.cc]: use a reference for this std::vector

**Comment by tjhei** [comment]
Jonathan, can you rebase to master so we can take a look at the test diffs please?

File: source/material_model/composition_reaction.cc (hunk 1 above)
Comment (bangerth) [inline source/material_model/composition_reaction.cc]: Or just use the original variables. It's not like these are much longer :-)

**Comment by bangerth** [comment]
I wanted to merge this (it looks good to me), but realized that it had one failing test on the regression test linked to above.

There are currently no tests listed on the tester website at http://cdash.kyomu.43-1.org/index.php?project=aspect&display=project so I can not say whether that test failed without the patch, or whether the patch here is at fault.

@tjhei : Can you run this PR on your tester (and without the PR) and see if it causes the regression? Unfortunately, all of the `composition_reaction` tests seem to fail on my machine...

**Comment by tjhei** [comment]
I looked into it and there was indeed an issue with the test, see #580

**Pull request closed without merging**